lib/modules/asset/AssetService.ts

```diff
@@ -279,239 +279,157 @@ export class AssetService {
-          );
-        }
+      if (engine.group !== newEngine.group) {
+        throw new BadRequestError(
+          `Engine ${newEngineId} is not in the same group as ${engineId}`
+        );
+      }
 
-        if (assetsList.length === 0) {
-          throw new BadRequestError("No assets to migrate");
-        }
+      //Get all assets to migrate
+      const assets = await this.sdk.document.mGet<AssetContent>(
+        engineId,
+        InternalCollection.ASSETS,
+        assetsList
+      );
+      errors.push(...assets.errors);
 
-        const assets = await this.sdk.document.mGet<AssetContent>(
-          engineId,
-          InternalCollection.ASSETS,
-          assetsList
-        );
+      if (assets.successes.length === 0) {
+        this.context.log.error("No assets found to migrate");
+        return { errors, successes };
+      }
 
-        // check if the assets exists in the other engine
-        const existingAssets = await this.sdk.document.mGet<AssetContent>(
-          newEngineId,
-          InternalCollection.ASSETS,
-          assetsList
-        );
+      //Get all the asset content, in order to create thm by batch
+      const assetsContent = assets.successes.map((asset) => ({
+        _id: asset._id,
+        body: asset._source,
+      }));
+
+      //We want to create the new asset with linked devices and groups empty
+      const assetsContentCopy = _.cloneDeep(assetsContent);
+      for (const asset of assetsContentCopy) {
+        asset.body.linkedDevices = [];
+        asset.body.groups = [];
+      }
 
-        if (existingAssets.successes.length > 0) {
-          throw new BadRequestError(
-            `Assets ${existingAssets.successes
-              .map((asset) => asset._id)
-              .join(", ")} already exists in engine ${newEngineId}`
+      //Whateever they exist in the destination tenant, try to create them all,
+      //using batch
+      const assetsCreated = await this.sdk.document.mCreate(
+        newEngineId,
+        InternalCollection.ASSETS,
+        assetsContentCopy
+      );
+      //We consider here we will return as success what we have been able
+      //to create, and related errors
+      const assetsCreatedId = assetsCreated.successes.map((a) => a._id);
+      const assetsNotCreatedId = assetsCreated.errors.map(
+        (a) => a.document._id
+      );
+      successes.push(...assetsCreatedId);
+      errors.push(...assetsNotCreatedId);
+
+      //Iterate over all created asset, and migrate each one
+      for (const asset of assetsCreated.successes) {
+        //We need to recover the linked devices to this asset,
+        //because we reset them when we create the asset
+        const assetOriginal = assets.successes.find((a) => a._id === asset._id);
+        if (!assetOriginal) {
+          this.context.log.error(
+            `Impossible to recover the original asset with id = ${asset._id}`
           );
+          continue;
         }
```

Comment:
This condition is never reachable since `assets.successes` contains all the assets that you've get before, and `assetsCreated` is the result of the call to `mCreate` based on the data obtained with `assets.successes` meaning that if an assets as successfuly been created it will always be contained  in the `assets.successes` thus assetOriginal is always going to exists and this condition is not useful.

Reply:
Yes, indeed. I know... but as it can programmatically reach undefined, I have keep this test it anyway.
I can remove it.